components/brave_rewards/browser/rewards_notification_service_impl.cc

```diff
@@ -313,9 +322,16 @@ void RewardsNotificationServiceImpl::OnGrantFinish(
     RewardsService* rewards_service,
     unsigned int result,
     Grant grant) {
-  DeleteNotification("rewards_notification_grant_" + grant.promotionId);
+  std::string grant_type = grant.type;
+  std::string prefix = grant_type == "ugp"
```

Comment:
this duplication should be refactored into a method

Reply:
done